gst/nnstreamer/tensor_filter/tensor_filter.c

```diff
@@ -110,58 +110,15 @@
 
 #define g_free_const(x) g_free((void*)(long)(x))
 
-typedef struct _TensorFilterSPList TensorFilterSPList;
-/**
- * @brief Linked list having all registered filter subplugins
- */
-struct _TensorFilterSPList
-{
-  TensorFilterSPList *next; /**< "Next" in the list */
-  GstTensorFilterFramework *body; /**< "Data" in the list element */
-};
-static GstTensorFilterFramework unknown = {
-  .name = "unknown",
-};
-static TensorFilterSPList listhead = {.next = NULL,.body = &unknown };
-
 /**
  * @brief Filter subplugin should call this to register itself
  * @param[in] tfsp Tensor-Filter Sub-Plugin to be registered
  * @return TRUE if registered. FALSE is failed or duplicated.
  */
-gboolean
+int
 tensor_filter_probe (GstTensorFilterFramework * tfsp)
 {
-  TensorFilterSPList *list;
-
-  if (!tfsp || !tfsp->name || !tfsp->name[0]) {
-    GST_ERROR ("Cannot register invalid filter subplugin.\n");
-    return FALSE;
-  }
-
-  list = &listhead;
-  do {
-    if (0 == g_strcmp0 (list->body->name, tfsp->name)) {
-      /* Duplicated! */
-      GST_ERROR ("DUplicated filter sub-plugin name found: %s\n", tfsp->name);
-      return FALSE;
-    }
-    if (list->next == NULL) {
-      TensorFilterSPList *next = g_malloc (sizeof (TensorFilterSPList));
-      next->next = NULL;
-      next->body = tfsp;
-      list->next = next;
-      break;
-    }
-    list = list->next;
-  } while (list != NULL);
-
-  GST_INFO ("A new sub-plugin, \"%s\" is registered for tensor_filter.\n",
-      tfsp->name);
-
-  /** @todo @bug unregister at exit */
-  hold_register_subplugin (NNS_SUBPLUGIN_FILTER, tfsp->name, tfsp);
-
+  register_subplugin (NNS_SUBPLUGIN_FILTER, tfsp->name, tfsp);
```

Comment:
tensor_filter_probe always returns TRUE whatever register_subplugin returns. How about returning the result of register_subplugin for the return value of the probe function.

Reply:
Yup. it returns FALSE if duplicated. Will update.